### PR TITLE
Set `baselineCapabilitySet: v4.11` by default for OCP 4.11+ 

### PR DESCRIPTION
**Pull request by simu**
OpenShift 4.11 has introduced optional cluster capabilities which can be configured, cf. [the upstream docs]. We update the component to set the `v4.11` baseline capability set by default on OCP 4.11+. This way we ensure that additional capabilities of future versions aren't enabled by default during cluster upgrades.

To determine a cluster's OCP version, we introduce a new component parameter which users should set to the cluster's dynamic fact `openshiftVersion`.

[the upstream docs]: https://docs.openshift.com/container-platform/4.11/post_installation_configuration/cluster-capabilities.html

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: component/main.jsonnet [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 
+local cluster_gt_411 =
+  params.openshiftVersion.Major == '4' &&
+  std.parseInt(params.openshiftVersion.Minor) >= 11;
+
 local clusterVersion = kube._Object('config.openshift.io/v1', 'ClusterVersion', 'version') {
-  spec: params.spec,
+  spec: {
+    [if cluster_gt_411 then 'capabilities']: {
+      baselineCapabilitySet: 'v4.11',
```
Comment (corvus-ch) [inline component/main.jsonnet:16]: Might be worth to have this exposed as a parameter.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -13,18 +28,39 @@ default::
...
 ----
 
-See the https://docs.openshift.com/container-platform/latest/updating/updating-cluster-between-minor.html[OpenShift docs] for available paramters and values.
+See the https://docs.openshift.com/container-platform/latest/updating/updating-cluster-between-minor.html[OpenShift docs] for available parameters and values.
+
+[NOTE]
+====
+Field `clusterID` must be set in the `ClusterVersion` object.
+The value for this field is supposed to be extracted from the cluster as a fact.
+However, the corresponding dynamic fact isn't implemented yet.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:40]: Is there an issue to have this addressed?

**Review by corvus-ch — CHANGES_REQUESTED** [review]
It probably is worth to have the baseline capabilities configurable. Also, there is the option to set additional capabilities. Can you please expose them, so we have that available in future should we need it?

Also is there an issue to address the missing cluster ID fact?

File: component/main.jsonnet (hunk 1 above)
Comment (simu) [reply component/main.jsonnet:16]: You can already override this via `parameters.openshift4_version.spec.capabilities.baselineCapabilitySet` the same as all the other available fields of `ClusterVersion`.

**Review by corvus-ch — APPROVED** [review]

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 2 above)
Comment (simu) [reply docs/modules/ROOT/pages/references/parameters.adoc:40]: I created an issue now [OCP-491 (internal)](https://ticket.vshn.net/browse/OCP-491)

**Pull request merged**